packages/shell-api/src/streams.ts

```diff
@@ -0,0 +1,172 @@
+import type { Document } from 'bson';
+import { CommonErrors, MongoshInvalidInputError } from '@mongosh/errors';
+
+import { MongoshInterruptedError } from './interruptor';
+import {
+  ShellApiWithMongoClass,
+  returnsPromise,
+  shellApiClassDefault,
+} from './decorators';
+import type ShellInstanceState from './shell-instance-state';
+import StreamProcessor from './stream-processor';
+import { ADMIN_DB, asPrintable } from './enums';
+import type Mongo from './mongo';
+
+@shellApiClassDefault
+export class Streams extends ShellApiWithMongoClass {
+  public static newInstance(instanceState: ShellInstanceState) {
+    return new Proxy(new Streams(instanceState), {
+      get(target, prop) {
+        const v = (target as any)[prop];
+        if (v !== undefined) {
+          return v;
+        }
+        if (typeof prop === 'string') {
```

Comment:
Can we add a `prop.startsWith('_')` check here similar to the ones for Database/Collection? That gives us the ability to add underscore-prefixed private properties without a chance of conflicts with user code.

Reply:
This is guarded by the previous if statement. Things on the `target` itself take precedence.

Reply:
Not quite, though. The reason that Database & Collection have these properties is future-proofing: Right now, `sp._foo` would return the `_foo` stream processor, but if we ever add a `_foo` private property, that would break user code which relies on the previous behavior. If we do not allow accessing underscore-prefixed stream processors this way from the beginning, that won’t happen.

Reply:
Right, I see what you mean now. Will add the check.